File: backend/src/main/java/com/gdje_izlazimo/project/dto/request/create/CreateEventRequest.java
package com.gdje_izlazimo.project.dto.request.create;

import com.gdje_izlazimo.project.enums.EventType;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record CreateEventRequest(

        UUID venueId,

        @NotBlank(message = "Event name is required")
        @Size(max = 150, message = "Event name must not exceed 150 characters")
        String name,

        String description,

        @NotNull(message = "Event date/time is required")
        @Future(message = "Event date/time must be in the future")
        LocalDateTime eventDateTime,

        LocalDateTime eventEndDateTime,

        String locationName,
        String locationAddress,

        EventType eventType,

        String externalOrganizerName,
        String externalOrganizerInstagram,

        Boolean featured,

        Double latitude,
        Double longitude,

        @Valid
        List<EventTicketTypeRequest> ticketTypes
) {}

File: frontend/src/app/core/models/events/create-event.request.ts
import { EventTicketTypeDto } from './event-response.dto';

export interface EventTicketTypeRequest {
  name: string;
  description?: string;
  price?: number | null;
  currency?: string;
  purchaseUrl: string;
  displayOrder?: number;
  active?: boolean;
}

export interface CreateEventDto {
  venueId?: string | null;
  name: string;
  description?: string;
  eventDateTime: string;
  eventEndDateTime?: string | null;
  locationName?: string | null;
  locationAddress?: string | null;
  eventType?: string | null;
  externalOrganizerName?: string | null;
  externalOrganizerInstagram?: string | null;
  featured?: boolean;
  imageUrl?: string;
  ticketTypes?: EventTicketTypeRequest[];
  latitude?: number | null;
  longitude?: number | null;
}